src/models/arellano_vfi.jl

```diff
@@ -0,0 +1,333 @@
+        end
+    end
+
+    nothing
+end
+
+"""
+This function takes the Arellano economy and its value functions and
+policy functions and then updates the prices for each (y, B') pair
+
+##### Arguments
+
+* `ae::ArellanoEconomy` : This is the economy we would like to update the
+                          prices for
+
+##### Notes
+
+* This function updates the prices and default probabilities in place
+"""
+function compute_prices!(ae::ArellanoEconomy)
+    # Unpack parameters
+    β, γ, r, ρ, η, θ, ny, nB = _unpack(ae)
+
+    # Create default values with a matching size
+    vd_compat = repmat(ae.vd, nB)
+    default_states = vd_compat .> ae.vc
+
+    # Update default probabilities and prices
+    copy!(ae.defprob, default_states * ae.Π')
+    copy!(ae.q, (1 - ae.defprob) / (1 + r))
+
+    nothing
+end
+
+"""
+This performs value function iteration and stores all of the data inside
+the ArellanoEconomy type.
+
+##### Arguments
+
+* `ae::ArellanoEconomy` : This is the economy we would like to solve
+* `tol::Float64` : Level of tolerance we would like to achieve
+* `maxit::Int` : Maximum number of iterations
+
+##### Notes
+
+* This updates all value functions, policy functions, and prices in place.
+
+"""
+function vfi!(ae::ArellanoEconomy; tol=1e-8, maxit=10000)
+
+    # Unpack stuff
+    β, γ, r, ρ, η, θ, ny, nB = _unpack(ae)
+    ygrid, ydefgrid, Bgrid, Π, vf, vd, vc, policy, q, defprob = _unpackgrids(ae)
+    Πt = Π'
+
+    # Iteration stuff
+    it = 0
+    dist = 10.
+
+    # Allocate memory for update
+    V_upd = zeros(ae.vf)
+
+    while dist > tol && it < maxit
+        it += 1
+
+        # Compute expectations for this iterations
+        # (We need Π' because of order value function dimensions)
+        copy!(V_upd, ae.vf)
+        EV = ae.vf * Πt
+        EVd = ae.vd * Πt
+        EVc = ae.vc * Πt
+
+        # Update Value Function
+        one_step_update!(ae, EV, EVd, EVc)
+
+        # Update prices
+        compute_prices!(ae)
+
+        dist = maxabs(V_upd - ae.vf)
+
+        if it%25 == 0
+            println("Finished iteration $(it) with dist of $(dist)")
+        end
+    end
+
+    nothing
+end
+
+"""
+This function simulates the Arellano economy
+
+##### Arguments
+
+* `ae::ArellanoEconomy` : This is the economy we would like to solve
+* `T::Int` : Number of periods to simulate
+* `y_init::Float64` : The level of income we would like to start with
+* `B_init::Float64` : The level of asset holdings we would like to start with
+
+##### Returns
+
+* `B_sim_val::Vector{Float64}` : Simulated values of assets
+* `y_sim_val::Vector{Float64}` : Simulated values of income
+* `q_sim_val::Vector{Float64}` : Simulated values of prices
+* `default_status::Vector{Float64}` : Simulated default status (True if in default)
+
+##### Notes
+
+* This updates all value functions, policy functions, and prices in place.
+
+"""
+function QuantEcon.simulate(ae::ArellanoEconomy, T::Int=5000;
```

Comment:
Let's rename `T` to `capT`. As @ZacCranko once pointed out, there is a very strong convention to have `T` be a type parameter in Julia code